nixos/modules/services/matrix/synapse.nix

```diff
@@ -158,12 +174,33 @@ in {
         '';
       };
 
+      extras = mkOption {
+        type = types.listOf (types.enum (lib.attrNames pkgs.matrix-synapse.optional-dependencies));
+        default = [
+          # Support internationalized domain names in user-search
+          "user-search"
+          # Provide the JournalHandler for the default log_config
+          "systemd"
+        ];
+        example = literalExpression ''
+          [
+            # Provide statistics about caching memory consumption
+            "cache-memory"
```

Comment:
No strong opinion. We could make it conditional on some indicator for prometheus usage.

Reply:
This seems to require a nested lookup for certain listener resources, which makes this a bit annoying to discover. Would like to consider it out of scope for now.

Reply:
We can't reliable automatically discover this since prometheus scraping is often done from another machine and can also be configured in multiple, diverse ways.